crates/common/download/src/download.rs

```diff
@@ -175,8 +150,13 @@ impl Downloader {
         let tmp_target_path = self.temp_filename().await?;
         let target_file_path = self.target_filename.as_path();
 
-        let mut file: File = File::create(&tmp_target_path)
-            .context(format!("Can't create a temporary file {tmp_target_path:?}"))?;
+        let temp_dir = self
+            .target_filename
```

Comment:
It's not really related to your PR, however, I got confused by the name `target_filename`. At first I thought it's just a name, but the type is actualy `PathBuf`. I wish it was `target_filepath` or so.